### PR TITLE
feat(nargo): Default expression width field in `Nargo.toml`

### LINKED ISSUE #4961: Add default `expression-width` field to `Nargo.toml`
As part of #4960 we're no longer going to be able to query the backend to see the correct expression width to use. We can still use 4 as a default but we're going to want to be able to set a new default with a key in `Nargo.toml` to avoid users using backends other than `bb` from having to specify it each time.

This entry should be ignored in any library crates. Also need to consider how workspaces are affected.

### PR DESCRIPTION
**Pull request by vezenovm**
# Description

## Problem\*

Resolves #4961 

## Summary\*

This PR let's a developer override the default expression width target with a field in the `Nargo.toml` manifest. 

Due to the limitations of the `clap` macros I had to switch `expression_width` on the `CompileOptions` struct to optional. I then hardcoded a constant that specifies we want width 4 as Noir's compilation default. We then settle on a target expression width by checking the following:
1. Is there a user supplied expression width from the CLI flag? If there is use that width.
2. Is there no user supplied width and an expression width specified in the manifest? Use the expression width in the manifest.
3. Is there no user supplied width and no width specified in the manifest? Use the hardcoded default value.

## Documentation\*

Check one:
- [ ] No documentation needed.
- [x] Documentation included in this PR.
- [ ] **[For Experimental Features]** Documentation to be submitted in a separate PR.

# PR Checklist\*

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

File: compiler/wasm/src/compile.rs [hunk 1]
```diff
@@ -164,10 +164,9 @@ pub fn compile_program(
     console_error_panic_hook::set_once();
     let (crate_id, mut context) = prepare_context(entry_point, dependency_graph, file_source_map)?;
 
-    let compile_options = CompileOptions {
-        expression_width: ExpressionWidth::Bounded { width: 4 },
-        ..CompileOptions::default()
-    };
+    let expression_width = ExpressionWidth::Bounded { width: 4 };
```
Comment (TomAFrench) [inline compiler/wasm/src/compile.rs:167]: Can you open an issue for us to make this customisable? Not a blocker for this PR as expression width was already fixed.

**Review by TomAFrench — APPROVED** [review]
I think the test doesn't really add much imo but otherwise looks good.

**Comment by vezenovm** [comment]
> I think the test doesn't really add much imo but otherwise looks good.

Yeah, I just thought maybe to show different gate counts with the different widths.

File: compiler/wasm/src/compile.rs (hunk 1 above)
Comment (vezenovm) [reply compiler/wasm/src/compile.rs:167]: https://github.com/noir-lang/noir/issues/5509

**Pull request merged**